src/lib/search/sync.ts

```diff
@@ -32,243 +31,81 @@ const sync = async () => {
-    const documents = i.documents;
-    await waitForTasks(
-      () => [index.addDocuments(documents, { primaryKey: "id" })],
-      `Adding data to ${index.uid}`,
-    );
-  }
-
-  await waitForTasks(
-    () => allIndexes.map((index) => index.resetSearchableAttributes()),
-    "Resetting searchable attributes",
-  );
-
-  await waitForTasks(
-    () =>
-      indexWithData.map((i) =>
-        i.index.updateSearchableAttributes(i.searchableAttributes),
-      ),
-    "Updating searchable attributes",
-  );
-
-  for (const i of indexWithData) {
-    if (i.sortableAttributes?.length) {
-      await waitForTasks(
-        () => [i.index.resetSortableAttributes()],
-        `Resetting sortable attributes for ${i.index.uid}`,
-      );
-      await waitForTasks(
-        () => [i.index.updateSortableAttributes(i.sortableAttributes)],
-        `Updating sortable attributes for ${i.index.uid}`,
-      );
-    }
-    if (i.rankingRules?.length) {
-      await waitForTasks(
-        () => [i.index.resetRankingRules()],
-        `Resetting ranking rules for ${i.index.uid}`,
-      );
-      await waitForTasks(
-        () => [i.index.updateRankingRules(i.rankingRules)],
-        `Updating ranking rules for ${i.index.uid}`,
-      );
-    }
-    if (i.typoTolerance !== undefined) {
-      await waitForTasks(
-        () => [i.index.resetTypoTolerance()],
-        `Resetting typo tolerance for ${i.index.uid}`,
-      );
-      await waitForTasks(
-        () => [i.index.updateTypoTolerance(i.typoTolerance)],
-        `Updating typo tolerance for ${i.index.uid}`,
+    for (const index of availableSearchIndexes) {
+      await waitForTask(
+        () => meilisearch.createIndex(index),
+        `Creating index ${index}`,
       );
     }
+    meiliInitialized = true;
   }
 
+  await syncMembers();
+  await syncSongs();
+  await syncArticles();
+  await syncEvents();
+  await syncPositions();
+  await syncCommittees();
+
   console.log(`Meilisearch: Data synced. Took ${Date.now() - currentTime} ms`);
-  return JSON.stringify(indexWithData);
+  return "Data synced";
 };
 
 export default sync;
 
-// HELPER FUNCTIONS
-async function waitForTasks(
-  fn: () => Array<Promise<EnqueuedTask>>,
-  taskName: string,
-  timeOutMs = 60 * 1000,
-) {
-  const currentTime = Date.now();
-  console.log(`Meilisearch: Waiting for "${taskName}" to finish`);
-  const enqueded = await Promise.all(fn());
-  const taskUids = enqueded.map((task) => task.taskUid);
-  const tasks = await meilisearch.waitForTasks(taskUids, { timeOutMs });
-  console.log(
-    `Meilisearch: "${taskName}" finished in ${Date.now() - currentTime} ms`,
-  );
-  return tasks;
-}
+/**
+ * Meilisearch has constraints on the id field:
+ * it can only contain alphanumeric characters (a-z, A-Z, 0-9), hyphens (-), and underscores (_).
+ * So we need to convert the prisma id to a meilisearch id.
+ * As of now, the Positions table has a prisma id that is not compatible with meilisearch.
+ * (it contains dots)
+ */
```

Comment:
Is this comment supposed to be here as well?

Reply:
It doesn't have to be there, it's more of a relic from when I used `uuid` instead. But since there is an explanatory comment documenting the converting funciton, this one could be removed